### PR TITLE
Storio sqlite object mapping

### LINKED ISSUE #278: Add transaction protection for DefaultPutResolver to avoid concurrency problems

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
`StorIOSQLite` now has new scheme of Object Mapping and Operations Resolvers as we discussed in #274 

Also, it closes #278 (Transaction for Put Operation Default Resolver)

@nikitin-da PTAL

1420 additions, 1670 deletions, as you can see, new scheme requires 250 lines less than old :)

### CODE REVIEW

File: storio-sqlite/src/androidTest/java/com/pushtorefresh/storio/sqlite/impl/QueryTest.java [hunk 1]
```diff
@@ -265,23 +290,16 @@
...
         final List<User> users = TestFactory.newUsers(20);
 
         int counter = 1;
-        for (User user : users) {
 
+        for (int i = 0; i < users.size(); i++) {
             char[] chars = new char[counter++];
-            Arrays.fill(chars, '*');
-            user.setEmail(new String(chars));
+            Arrays.fill(chars, '*'); // wtf is going on?
```
Comment (nikitin-da) [inline storio-sqlite/src/androidTest/java/com/pushtorefresh/storio/sqlite/impl/QueryTest.java]: Is it question for me?

File: storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/design/PutOperationDesignTest.java [hunk 2]
```diff
@@ -12,8 +17,34 @@
...
+        protected UpdateQuery mapToUpdateQuery(@NonNull ContentValues contentValues) {
+            return new UpdateQuery.Builder()
+                    .table("some_table") // it's just a sample, no need to specify params
+                    .build();
+        }
+
+        @NonNull
+        @Override
+        protected ContentValues mapToContentValues(@NonNull ContentValues contentValues) {
+            return contentValues; // easy
```
Comment (nikitin-da) [inline storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/design/PutOperationDesignTest.java]: Hm

**Comment by nikitin-da** [comment]
You write faster then I read.

File: storio-sqlite/src/androidTest/java/com/pushtorefresh/storio/sqlite/impl/QueryTest.java (hunk 1 above)
Comment (artem-zinnatullin) [inline storio-sqlite/src/androidTest/java/com/pushtorefresh/storio/sqlite/impl/QueryTest.java]: Haha, no :) I remembered that test but I had to spend couple of minutes to understand what's going on

File: storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/design/PutOperationDesignTest.java (hunk 2 above)
Comment (artem-zinnatullin) [inline storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/design/PutOperationDesignTest.java]: Yep, such easy

File: storio-sqlite/src/androidTest/java/com/pushtorefresh/storio/sqlite/impl/QueryTest.java (hunk 1 above)
Comment (nikitin-da) [inline storio-sqlite/src/androidTest/java/com/pushtorefresh/storio/sqlite/impl/QueryTest.java]: So we need better comment or/and more obvious variable name

2015-04-15 13:07 GMT+03:00 Artem Zinnatullin <notifications@github.com>:

**Pull request merged**